Database-Features/removing-corrupt-data.md

```diff
@@ -0,0 +1,91 @@
+4. Re-create the table using the saved DDL.
+5. Reload the data into the table from another source. 
+
+If one of the statistics tables fails, contact Exasol to delete the corresponding data. Once all of the corrupt objects are dropped, re-run the above script to make sure there are no errors. 
+
+If the database is still crashing after all of the tables are fixed, you can try to drop all of the indexes in case one of them was also corrupted. The steps for this are:
+
+1. (optional) Run the below query to generate ENFORCE INDEX statements.
+```sql
+select 'ENFORCE '||case when IS_LOCAL then 'LOCAL' else 'GLOBAL' end||' INDEX ON "'||INDEX_SCHEMA||'"."'||INDEX_TABLE||'" ('||GROUP_CONCAT(('"'||COLUMN_NAME||'"')  order by
+ordinal_position)||');' from "$EXA_INDEX_COLUMNS" where INDEX_TABLE not like 'RPL:%'  group by index_object_id, index_schema, index_table, is_local order by COUNT(*) ASC;
+```
+2. (optional) Save the results to a separate text file
+3. Run the below query to generate DROP INDEX statements
+```sql
+select 'DROP '||case when IS_LOCAL then 'LOCAL' else 'GLOBAL' end||' INDEX ON "'||INDEX_SCHEMA||'"."'||INDEX_TABLE||'" ('||GROUP_CONCAT(('"'||COLUMN_NAME||'"')  order by
+ordinal_position)||');' from "$EXA_INDEX_COLUMNS" where INDEX_TABLE not like 'RPL:%'  group by index_object_id, index_schema, index_table, is_local order by COUNT(*) ASC;
+```
+4. Save the results to a separate text file
+5. Run the generated queries which perform all of the DROP INDEX statements
+6. (optional) Run the generated queries which perform all of the ENFORCE INDEX statements.
+
+Note - the database will automatically create indexes as needed, therefore the ENFORCE INDEX statements are purely optional and should only be run after you verify that the database is behaving normal. 
+
+**If after all of the above actions, the database is still restarting during query execution, then this may point to the fact that an underlying data structure was corrupted and the database is unrecoverable. In this case, the only option would be to delete the database, re-create it, and reload all of the data.**
+
+### 3. Re-create Database from Metadata
+
+If all of the data is easily restorable from different sources and there is no valid remote backup present, the fastest option may be to save the Metadata (DDL of all objects) of the database, delete the database, create a new database, restore the metadata, and reload all of the data. Exasol provides a [script](https://raw.githubusercontent.com/exasol/exa-toolbox/master/utilities/create_db_ddl.sql) to save the metadata of the entire database. In this script, all schemas, tables, views, scripts, functions, users, roles, permissions, and connections are created. However, any user or connection passwords are lost and must be reset afterwards. For more information, see [Create DDL for the entire database](create-ddl-for-the-entire-database.md).
+
+The exact steps would be:
```

Comment:
Let's remove this, or put it under some different label / topic.

"how to replicate my database using SQL scripts"

Reply:
We already have this, so I removed some of the steps and just left the paragraph.